### ISSUE TITLE
Current execution row is not highlighting in debug mode after last update

### ISSUE THREAD

**Issue by zhelnio** [microsoft/vscode-cpptools#526]
problem screen:
It is stopping on breakpoint at line 73, but this line is not highlighted
![2017-03-03_00-44-47](https://cloud.githubusercontent.com/assets/6334288/23528476/16c1905e-ffab-11e6-951f-9d0e6f644e95.png)
version info:
![2017-03-03_00-46-20](https://cloud.githubusercontent.com/assets/6334288/23528479/186134e6-ffab-11e6-8ed2-54e83fd13bfd.png)

**Comment by pieandcakes** [comment]
@zhelnio Which OS is this on? I tried it on Ubuntu 14.04 with VSCode `1.10.1` the latest `0.10.2` extension and a C file and a breakpoint and I don't see this at all. Will need more information.

**Comment by zhelnio** [comment]
Windows 7 SP1 x64. 
I'm using it for remote debug.
VSCode config: https://github.com/zhelnio/memos/tree/master/public/00_mips_ftdi_debug/src
It worked good about day or two ago, and its behavior changed after last update.

**Comment by Valmadson** [comment]
in my notebook, too not highlighted in debug.

vscode 1.9
linux 32

**Comment by UR3IQO** [comment]
Windows 8 x64 and VSCode 1.10.1/cpptools 0.10.2 does not highlight current line too. It was ok before the last update. Unfortunately I have updated both VSCode and cpptools, so it is unclear whether this problem caused by VSCode update or cpptools update.

**Comment by sean-mcmanus** [comment]
The bug repros for me too. Stepping will get the line to highlight but all further stepping doesn't work unless a breakpoint is hit.

**Comment by pieandcakes** [comment]
@sov1178 @zhelnio I'm wondering if you step back to VSCode 1.9 if that works? Or can either of you give me a good repro application since I can't seem to reproduce it. 

@Valmadson 's issue is different and documented in #520 
@sean-mcmanus is working on Windows with our Visual C++ debugger and is also a separate issue.

**Comment by zhelnio** [comment]
@pieandcakes if I'll give you some access to my PC could it help to find the problem?

**Comment by pieandcakes** [comment]
I would need to be able to debug it with debug bits. if you do a simple hello world project or a basic threading project does it reproduce for you?

**Comment by zhelnio** [comment]
@pieandcakes Well....I have something like this: a small remote debug project for the MIPSfpga SoC. Send me some mail (my email is in my profile) and I'll give you the  join.me link or anydesk.com access

**Comment by pieandcakes** [comment]
@zhelnio Sure. Let me send you some steps to get diagnostic logging first so I can try that before we try connecting to your machine.

**Comment by zhelnio** [comment]
@pieandcakes Ok, I am ready

**Comment by pieandcakes** [comment]
@zhelnio Thank you. I have submitted a fix: https://github.com/Microsoft/MIEngine/pull/565 for the issue I see with your log. 

Once it is approved I will work on getting a build out to you to test. 

@sov1178 If you can enable `"logging": { "engineLogging": true } in your `launch.json` and retry and send me the block that is sent in the `thread-info` call, I can confirm that it is the same issue that you are running into.

**Comment by UR3IQO** [comment]
@pieandcakes I have rollback the cpptools to 0.10.1 and problem disappeared. So I can confirm the problem is in cpptools extension.

I have enabled logging and found  `thread-info` call (it was called once during initialization):
```
1: (2172) <-1014-thread-info
1: (2181) ->1014^done,threads=[{id="1",target-id="Remote target",frame={level="0",addr="0x002001d0",func="Reset_Handler",args=[],file="Startup.c",fullname="d:\\Oleg\\!Projects\\HAMHard\\TrcrvNew-(Neon)\\Soft\\Ctrl2\\Ctrl\\Startup.c",line="332"},state="stopped"}],current-thread-id="1"
1: (2181) ->(gdb)
```
Here is a piece of the log with the exchange on breakpoint hit:
```
Breakpoint 2, entryDisplayTask2 () at Display.c:892
1: (4514) ->~"892\t            TFTSetDC(&menudc[vpopup]);\n"
892	            TFTSetDC(&menudc[vpopup]);
1: (4514) ->*stopped,reason="breakpoint-hit",disp="keep",bkptno="2",frame={addr="0x00205b0a",func="entryDisplayTask2",args=[],file="Display.c",fullname="d:\\Oleg\\!Projects\\HAMHard\\TrcrvNew-(Neon)\\Soft\\Ctrl2\\Ctrl\\Display.c",line="892"},thread-id="1",stopped-threads="all"
1: (4514) <-1017-stack-list-frames 0 1000
1: (4527) ->1017^done,stack=[frame={level="0",addr="0x00205b0a",func="entryDisplayTask2",file="Display.c",fullname="d:\\Oleg\\!Projects\\HAMHard\\TrcrvNew-(Neon)\\Soft\\Ctrl2\\Ctrl\\Display.c",line="892"},frame={level="1",addr="0x00000000",func="??"}]
1: (4527) ->(gdb)
1: (4527) 1017: elapsed time 13
Execute debugger commands using "-exec <command>", for example "-exec info registers" will list registers in use (when GDB is the debugger)
1: (4625) <-1018-stack-list-arguments 2 0 1
1: (4628) ->1018^done,stack-args=[frame={level="0",args=[]},frame={level="1",args=[]}]
1: (4628) ->(gdb)
1: (4630) 1018: elapsed time 5 
```
P.S. it appeared I cannot copy the whole debugger output to the clipboard (only a small piece copied) - it that another bug or feature?

**Comment by pieandcakes** [comment]
@sov1178 Thanks for the update.

You have the same issue that @zhelnio was having above where we have a bug when the `target-id` inside the `-thread-info` call is returning text that does not contain an actual Thread ID. 

I have a pull request out for the fix:
https://github.com/Microsoft/MIEngine/pull/565

I am hoping to release a small bugfix update this week to resolve this but I need to coordinate with the language server side first. This fix will be in the next release.

**Comment by pieandcakes** [comment]
@zhelnio @sov1178 This is fixed in `0.10.3` release. Please update the extension and if it is still happening comment/reopen this issue please. 

Thanks

**Comment by zhelnio** [comment]
@pieandcakes Perfect! It works good. Thank you!
